tests/dummy/app/templates/components/validated-input.hbs

```diff
@@ -5,10 +5,16 @@
   {{/if}}
 
   <div class="input-error">
-      {{#if showMessage}}
+      {{#if showErrorMessage}}
           <div class="error">
-              {{v-get model valuePath 'message'}}
+              {{get (v-get model valuePath 'error') 'message'}}
           </div>
       {{/if}}
+
+      {{#if showWarningMessage}}
+        <div class="warning">
+            {{get (v-get model valuePath 'warning') 'message'}}
+        </div>
+      {{/if}}
```

Comment:
I wonder if we should expose all the messages for an attribute somehow. Since this is new functionality, I wonder if `messages` should return all messages, including warnings. I could see wanting to do something like this for my flavor of validated input:
```handlebars
{{#each (v-get model valuePath 'messages') as |msg|}}
  <div class="message {{if msg.isWarning 'warning'}}">
     {{msg}}
  </div>
{{/each}}
```

That way I don't have to have two blocks and reduce boilerplate templating. What do you think?